File: services/core-go/internal/db/pool.go
package db

import (
	"context"
	"fmt"

	"github.com/jackc/pgx/v5/pgxpool"
)

// NewPool creates a pgxpool connection pool, verifies connectivity with Ping,
// and returns the pool ready to use. Caller is responsible for calling pool.Close().
func NewPool(ctx context.Context, dsn string) (*pgxpool.Pool, error) {
	cfg, err := pgxpool.ParseConfig(dsn)
	if err != nil {
		return nil, fmt.Errorf("db: parse dsn: %w", err)
	}

	cfg.MinConns = 2
	cfg.MaxConns = 10

	pool, err := pgxpool.NewWithConfig(ctx, cfg)
	if err != nil {
		return nil, fmt.Errorf("db: create pool: %w", err)
	}

	if err := pool.Ping(ctx); err != nil {
		pool.Close()
		return nil, fmt.Errorf("db: ping: %w", err)
	}

	return pool, nil
}

File: services/core-python/app/database/pool.py
import asyncpg


async def create_pool(dsn: str) -> asyncpg.Pool:
    """Create and return an asyncpg connection pool.

    Raises asyncpg.PostgresError if the DSN is invalid or the server is
    unreachable. Caller is responsible for calling pool.close() on shutdown.
    """
    return await asyncpg.create_pool(
        dsn,
        min_size=2,
        max_size=10,
    )
